streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -1537,4 +1545,25 @@ private RuntimeException performOnStreamTasks(final StreamTaskAction action) {
 
         return firstException;
     }
+
+    /**
+     * Return information about the current {@link StreamThread}.
+     *
+     * @return {@link ThreadMetadata}.
+     */
+    public final synchronized ThreadMetadata threadMetadata() {
```

Comment:
Instead of making these two functions synchronized, could we just declare `threadMetadata` as volatile since we are creating a new object each time?